### PR TITLE
Add built-in function cosh

### PR DESCRIPTION
**Pull request by dhruvpat3l**
**What type of PR is this?**

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [X] Feature
- [ ] Test and CI
- [ ] Code Refactoring

**Which issue(s) this PR fixes:**

issue #2095 

**What this PR does / why we need it:**

Add cosh() built-in function

**Special notes for your reviewer:**

Not Available

**Additional documentation (e.g. design docs, usage docs, etc.):**

Not Available

### CODE REVIEW

File: pkg/vectorize/utc_date/utc_date.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+var (
+	getUTCDate func() types.Date
+)
+
+func init() {
+	getUTCDate = GetUTCDate
+}
+
+func GetUTCDate() types.Date {
+	return types.Today()
```
Comment (broccoliSpicy) [inline pkg/vectorize/utc_date/utc_date.go:27]: by using the types.Today() function, I think you will get the local Date rather than the utc date.

**Review by broccoliSpicy — CHANGES_REQUESTED** [review]

**Review by domingozhang — APPROVED** [review]
bvt passed

File: pkg/vectorize/cosh/cosh.go [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+	coshUint16  func([]uint16, []float64) []float64
+	coshUint32  func([]uint32, []float64) []float64
+	coshUint64  func([]uint64, []float64) []float64
+	coshInt8    func([]int8, []float64) []float64
+	coshInt16   func([]int16, []float64) []float64
+	coshInt32   func([]int32, []float64) []float64
+	coshInt64   func([]int64, []float64) []float64
+	coshFloat32 func([]float32, []float64) []float64
+	coshFloat64 func([]float64, []float64) []float64
+)
```
Comment (broccoliSpicy) [inline pkg/vectorize/cosh/cosh.go:30]: perhaps it would be better if you follow the updated function variable initiation in pr#2225

**Review by broccoliSpicy — CHANGES_REQUESTED** [review]

**Comment by broccoliSpicy** [comment]
if we merge this pr before the utc_date pr, git may render the pr utc_date invalid as the PR utc_date won't have any modification compared to the main branch by then

File: pkg/vectorize/cosh/cosh.go (hunk 2 above)
Comment (dhruvpat3l) [reply pkg/vectorize/cosh/cosh.go:30]: So should "coshInt64", "coshFloat32", "coshFloat64" instead be "coshint64", "coshfloat32", "coshfloat64"

Comment (broccoliSpicy) [reply pkg/vectorize/cosh/cosh.go:30]: I think our function variable initiation now is like this
```go
var (
    CoshUint8
)

func init() {
    CoshUint8 = coshUint8
}
```

and Golang supports generic programming now, you can try using it in your cosh implementation if interested

**Comment by fengttt** [comment]
Close to free resource on ci.

**Pull request closed without merging**